swatch-producer-azure/src/main/java/com/redhat/swatch/azure/service/AzureMarketplaceClientFactory.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright Red Hat, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ * Red Hat trademarks are not licensed under GPLv3. No permission is
+ * granted to use or replicate Red Hat trademarks that are incorporated
+ * in this software or its documentation.
+ */
+package com.redhat.swatch.azure.service;
+
+import com.redhat.swatch.azure.file.AzureMarketplaceCredentials;
+import com.redhat.swatch.azure.file.AzureMarketplaceProperties;
+import com.redhat.swatch.azure.http.AzureMarketplaceHeaderProvider;
+import com.redhat.swatch.clients.azure.marketplace.api.resources.AzureMarketplaceApi;
+import io.quarkus.oidc.client.OidcClients;
+import jakarta.enterprise.context.ApplicationScoped;
+import jakarta.inject.Inject;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
+import lombok.extern.slf4j.Slf4j;
+import org.eclipse.microprofile.rest.client.RestClientBuilder;
+
+@Slf4j
+@ApplicationScoped
+public class AzureMarketplaceClientFactory {
+
+  AzureMarketplaceProperties azureMarketplaceProperties;
+  AzureMarketplaceCredentials azureMarketplaceCredentials;
+  private OidcClients oidcClients;
+
+  @Inject
+  public AzureMarketplaceClientFactory(
+      AzureMarketplaceProperties azureMarketplaceProperties,
+      AzureMarketplaceCredentials azureMarketplaceCredentials,
+      OidcClients oidcClients) {
+    this.azureMarketplaceProperties = azureMarketplaceProperties;
+    this.azureMarketplaceCredentials = azureMarketplaceCredentials;
+    this.oidcClients = oidcClients;
+  }
+
+  public List<AzureMarketplaceApi> createClientForEachTenant() {
+    return azureMarketplaceCredentials.getClients().stream()
+        .map(
+            client -> {
+              try {
+                return RestClientBuilder.newBuilder()
```

Comment:
Use QuarkusRestClientBuilder instead (this is the recommended builder for Quarkus Reactive REST Client).